src/agent/onefuzz-task/src/tasks/coverage/generic.rs

```diff
@@ -373,12 +373,19 @@ impl<'a> TaskContext<'a> {
             match entry {
                 Ok(entry) => {
                     if entry.file_type().await?.is_file() {
-                        self.record_input(&entry.path()).await?;
-                        count += 1;
-
-                        // make sure we save & sync coverage every 10 inputs
-                        if count % 10 == 0 {
-                            self.save_and_sync_coverage().await?;
+                        if let Err(e) = self.record_input(&entry.path()).await {
+                            warn!(
+                                "ignoring error recording coverage for input: {}, error: {}",
+                                entry.path().display(),
+                                e
+                            );
+                        } else {
+                            count += 1;
+
+                            // make sure we save & sync coverage every 10 inputs
+                            if count % 10 == 0 {
```

Comment:
Is this an expensive operation? It seems like an odd performance enhancement that might leave us with missing data if we don't wind up with an even 10 inputs.

Reply:
We do a final sync at the end; this is to ensure we have some data while coverage is running.